src/redux/middleware/resetSearchResults.tsx

```diff
@@ -0,0 +1,16 @@
+import { Middleware } from 'redux';
+
+export const searchResultsCleanUp: Middleware<any, any, any> = ({ dispatch }: any) => (next: any) => (action: any) => {
+  if (action.type === 'EMPTY_SEARCH_RESULT_STATE') {
+
+    next({
+      type: `EMPTY_SEARCH_RESULT_STATE`,
+    });
+    dispatch({
+      type: `${action.type}`,
+    })
+
+  } else {
+    return next(action);
+  }
+}
```

Comment:
is 'EMPTY_SEARCH_RESULT_STATE' being fired 3 times?

Reply:
no